File: src/main/java/br/com/blz/api/models/WarehouseEntity.java
package br.com.blz.api.models;

import java.io.Serializable;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.JoinColumns;
import javax.persistence.ManyToOne;
import javax.persistence.Table;
import javax.persistence.Transient;
import javax.validation.constraints.NotNull;

import org.springframework.beans.factory.annotation.Autowired;

@Entity
@Table(name = "warehouse")
public class WarehouseEntity implements Serializable {

	@Transient
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE)
	private long id;

	@NotNull
	private String locality;

	@NotNull
	private int quantity;

	@NotNull
	private String type;

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getLocality() {
		return locality;
	}

	public void setLocality(String locality) {
		this.locality = locality;
	}

	public int getQuantity() {
		return quantity;
	}

	public void setQuantity(int quantity) {
		this.quantity = quantity;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}
	

}

File: src/main/kotlin/com/boticario/testkotlin/infrastructure/persistence/entity/WarehouseEntity.kt
package com.boticario.testkotlin.infrastructure.persistence.entity

import com.boticario.testkotlin.domain.entity.Warehouse
import com.boticario.testkotlin.domain.enum.WarehouseType
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity
class WarehouseEntity(
        @Id
        @GeneratedValue(strategy = GenerationType.AUTO)
        private val id: Int = 0,
        private val locality: String = "",
        private val quantity: Int = 0,
        private val type: String = "",
        @ManyToOne
        @JoinColumn(name = "product_sku")
        private val product: ProductEntity? = null
) {
        fun toDomain() = Warehouse(
                locality,
                quantity,
                WarehouseType.valueOf(type)
        )

        companion object {
                fun fromDomain(warehouse: Warehouse) = WarehouseEntity(
                        locality = warehouse.locality,
                        quantity = warehouse.quantity,
                        type = warehouse.type.name
                )
        }
}